src/axom/spin/internal/linear_bvh/bvh_traverse.hpp

```diff
@@ -36,26 +36,42 @@ inline bool leaf_node(const int32& nodeIdx) { return (nodeIdx < 0); }
  * \param [in] p the primitive in query, e.g., a point, ray, etc.
  * \param [in] B functor that defines the check for the bins
  * \param [in] A functor that defines the leaf action
+ * \param [in] Comp functor used for determining which child node to traverse
+ *  down first if both bins are to be traversed
  *
  * \note The supplied functor `B` is expected to take the following two
  *  arguments:
  *    (1) The supplied primitive, p
  *    (2) a primal::BoundingBox< FloatType, NDIMS > of the BVH bin
  *
+ * \note The supplied functor `Comp` is expected to take the following three
+ *  arguments:
+ *    (1) The left child bounding box
+ *    (2) The right child bounding box
+ *    (3) The primitive being queried
+ *  It should return true if the right child is closer (indicating a swap is
+ *  necessary) and false if the left child is closer.
+ *
  * \see BVHData for the details on the internal data layout of the BVH.
  *
  * \note Moreover, the functor `B` returns a boolean status that indicates
  *  if the specified traversal predicate is satisfied.
  *
  */
-template <int NDIMS, typename FloatType, typename PrimitiveType, typename InBinCheck, typename LeafAction>
+template <int NDIMS,
+          typename FloatType,
+          typename PrimitiveType,
+          typename InBinCheck,
+          typename LeafAction,
+          typename TraversePref>
 AXOM_HOST_DEVICE inline void bvh_traverse(
   const primal::BoundingBox<FloatType, NDIMS>* inner_nodes,
   const int32* inner_node_children,
   const int32* leaf_nodes,
   const PrimitiveType& p,
   InBinCheck&& B,
-  LeafAction&& A)
+  LeafAction&& A,
+  TraversePref&& Comp)
```

Comment:
We are being rather inconsistent in casing in this function.

Reply:
I just went along with the different casing for the lambda arguments, but if there's another way we'd prefer to name them I'm all ears.